Исправление постраничной навигации

nook-ru opened the pull request:
Некорректные аргументы передавались в методы для получения параметров постранички (выдавался warning)

nook-ru committed:
Исправление постраничной навигации

Некорректные аргументы передавались в методы для получения параметров постранички (выдавался warning)

ASGAlex reviewed (commented):
CDBResult::GetNavParams может принимать первым аргументом массив. В данном случае он передается намеренно, чтобы в параметре sNavID указать ID пагинатора. Если посмотреть исходник битриксовой функции, видно, что там есть условие на этот случай. 

Если не сделать этого, то мы можем получить неприятный баг, когда на одной административной странице переключаем страницу, после чего все другие административные списки тоже начинают открываться с этой страницы. Если такой страницы вообще в БД нет, то можно было вообще схватить ни то фатал, ни то ексепшн.

А что за warning выскакивает?

nook-ru committed:
fix

https://github.com/DigitalWand/digitalwand.admin_helper/pull/61#pullrequestreview-18162291

nook-ru commented:
```
Illegal offset type in isset or empty (0)
\bitrix\modules\main\classes\general\user_options.php:111
#0: CUserOptions::GetOption(string, array)
	\bitrix\modules\main\interface\admin_lib.php:2444
#1: CAdminResult->GetNavSize(array)
	\local\vendor\digitalwand\digitalwand.admin_helper\lib\Helper\AdminListHelper.php:258
```

`CAdminResult::GetNavSize`() на массив не рассчитывает.
Но да, что-то я погорячился. Вернул `GetNavParams` с массивом.
